pkg/infinispan/grafana/dashboard_controller.go

```diff
@@ -0,0 +1,94 @@
+package grafana
+
+import (
+	"context"
+
+	"github.com/infinispan/infinispan-operator/pkg/controller/constants"
+	"github.com/infinispan/infinispan-operator/pkg/controller/infinispan"
+	grafanav1alpha1 "github.com/integr8ly/grafana-operator/v3/pkg/apis/integreatly/v1alpha1"
+	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
+	appsv1 "k8s.io/api/apps/v1"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/discovery"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var log = logf.Log.WithName("infinispan-grafana")
+
+func AddToScheme(scheme *runtime.Scheme) error {
+	return grafanav1alpha1.AddToScheme(scheme)
+}
+
+func Create(namespace string, mgr manager.Manager) error {
+	dc, err := discovery.NewDiscoveryClientForConfig(mgr.GetConfig())
+	if err != nil {
+		return err
+	}
+	// detect the GrafanaDashboard resource type resourceExists on the cluster
+	resourceExists, _ := k8sutil.ResourceExists(dc, grafanav1alpha1.SchemeGroupVersion.String(), grafanav1alpha1.GrafanaDashboardKind)
+	if !resourceExists {
+		log.Info("Grafana CRD not present - not installing dashboard CR")
+		return nil
+	}
+
+	// Using mgr.GetClient() does not work as we don't configure the cache for this scheme
+	controllerClient, err := client.New(mgr.GetConfig(), client.Options{})
+	if err != nil {
+		return err
+	}
+
+	infinispanDashboard := dashboard(namespace)
+
+	if _, err = controllerutil.CreateOrUpdate(context.Background(), controllerClient, infinispanDashboard, func() error {
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	operatorDeployment := &appsv1.Deployment{}
+	_, err = infinispan.LookupResource("infinispan-operator", namespace, operatorDeployment, controllerClient, log)
```

Comment:
Both `result` and `err` should be validated. If they are `nil` this means that Deployment is found
Also please pay attention that this logic should work even if Deployment not found (local run and test where Deployment entity is not present)

Reply:
Yeah, I didn't validate as I did it afterwards and didn't make much sense. So I will add the validation when I flip the ordering as it makes sense then ;)

Reply:
Updated to do the check before hand and added in return value checking. Need to still verify it works when there is a deployment though.